libr/bp/bp.c

```diff
@@ -409,3 +412,26 @@ R_API int r_bp_size(RBreakpoint *bp) {
 	}
 	return bpsize;
 }
+
+// Check if the breakpoint is in a valid map
+R_API bool r_bp_is_valid(RBreakpoint *bp, RBreakpointItem *b) {
+	RDebugMap *map;
+	RListIter *iter;
+	RList *maps;
+	if (!bp->bpinmaps) {
+		return true;
+	}
+
+	if (!(maps = bp->corebind.getDebugMaps (bp->corebind.core))) {
```

Comment:
Unnecessary if you see the code in is,apped younwill see that it already does all the debugger iomaps? Debug,aps checks

Reply:
Autocorrect butchered your sentence but I assume you meant to say that isMapped does the job?

In that case, it doesn't since it only checks if the address is in a map. It doesn't check the permissions as I wrote in a comment a few lines before. Plus, isMapped runs r_debug_maps_sync for each address which isn't desirable as we agreed earlier.